spec.md

```diff
@@ -15,4 +16,574 @@ The key words "unspecified", "undefined", and "implementation-defined" are to be
+* Define API (RPCs) that enable:
+  * Dynamic provisioning and de-provisioning of a bucket.
+  * Ability to define bucket access control mechanisms
+  * Consumption of both existing and new buckets created on SP.
+  * Define plugin protocol RECOMMENDATIONS.
+  * Describe a process by which a admin configures a Plugin.
+  * COSI deployments and templates. 
+
+### Non-Goals in MVP
+
+The Container Object Storage Interface (COSI) explicitly will not define, provide, or dictate:
+
+* Specific mechanisms by which a Plugin Sidecar manages the lifecycle of a Plugin, including:
+  * How to deploy, install, upgrade, uninstall, monitor, or respawn (in case of unexpected termination) Plugins.
+* A first class message structure/field to represent "grades of object storage" (aka "bucket class").
+* Protocol-level authentication and authorization.
+* Packaging of a Plugin.
+
+## Solution Overview
+
+This specification defines an interface along with the minimum operational and packaging recommendations for a storage provider (SP) to implement a COSI compatible plugin.
+The interface declares the RPCs that a plugin MUST expose: this is the **primary focus** of the COSI specification.
+Any operational and packaging recommendations offer additional guidance to promote cross-COSI system compatibility.
+
+### Architecture
+
+The primary focus of this specification is on the **protocol** between a COSI System and a Plugin.
+It SHOULD be possible to ship cross-COSI system compatible Plugins for a variety of deployment architectures.
+A COSI system SHOULD be equipped to handle both centralized and headless plugins.
+Several of these possibilities are illustrated in the following figures.
+
+```
+                    COSI System "Master" Host
++-------------------------------------------+
+|                                           |
+|  +------------+           +------------+  |
+|  |   COSI     |   gRPC    | Controller |  |
+|  |   System   +----------->   Plugin   |  |
+|  +------------+           +------------+  |
+|                                           |
++-------------------------------------------+
+
+```
+Figure 1: Plugin deployment, all of the COSI system Node hosts run Plugins. 
+
+
+![Cosi Components](cosicomponents.png)
+Figure 2: Interaction within COSI components 
+
+### Bucket Lifecycle
+
+```
+     CreateBucket +-------------+ DeleteBucket 
+    +------------->|  CREATED   +--------------+ 
+    |              +---+----^---+              |    
+    |          Grant   |    | Revoke           v    
+    +++        Bucket  |    | Bucket          +++  
+    |X|        Access  |    | Access          | |  
+    +-+            +---v----+---+             +-+  
+                   |   BOUND    |                  
+                   +---+----^---+                  
+
+```
+Figure 3: The lifecycle of a dynamically provisioned bucket, from
+creation to deletion.
```

Comment:
Unclear in diagram which steps are done by the user, which are done by an admin, and which are automated by COSI.

Reply:
The line under CreateBucket has an arrow pointing right. The line under DeleteBucket has no arrow. Is this intentional?

Reply:
The intention is to provide the lifecycle representation. Please add an` issue` for adding a section on roles and responsibilities for such operations.